pkg/dataencryption/dataencryption.go

```diff
@@ -0,0 +1,45 @@
+package dataencryption
+
+import (
+	"io"
+	"os"
+
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha1"
+	"crypto/sha256"
+
+	"github.com/gardener/etcd-backup-restore/pkg/types"
+)
+
+const dataEncryptionKeyEnvName = "DATA_ENCRYPTION_KEY"
+
+var dataEncryptionKey = []byte(os.Getenv(dataEncryptionKeyEnvName))
+
+func DecorateSnapStore(snapstore types.SnapStore) types.SnapStore {
+	if len(dataEncryptionKey) == 0 {
+		return snapstore
+	}
+
+	return &decoratedSnapStore{snapstore: snapstore}
+}
+
+func decorateReadCloser(snapName []byte, rc io.ReadCloser) io.ReadCloser {
+	iv := snapName // use snapName as the initial vector
+	stream := makeAesOfbStream(iv, dataEncryptionKey)
+	return &readerCloser{r: &cipher.StreamReader{S: stream, R: rc}, c: rc}
+}
+
+func makeAesOfbStream(iv []byte, key []byte) cipher.Stream {
+	_key := sha256.Sum256(key)
+
+	block, err := aes.NewCipher(_key[:])
+	if err != nil {
+		panic(err)
+	}
+
+	_iv := sha1.Sum(iv)
+
+	stream := cipher.NewOFB(block, _iv[:aes.BlockSize])
```

Comment:
Using OFB here makes the code very manageable (which is a positive!) but OFB in general is not a great encryption option for the real world: https://crypto.stackexchange.com/questions/26707/what-are-the-advantages-of-using-ofb-output-feedback-mode

Authenticated GCM seems more appropriate, since it's fast and secure.

Reply:
Thank you for the feedback! I agree. I will make changes to this PR next week.